cli/azd/pkg/project/scaffold_gen.go

```diff
@@ -186,7 +186,8 @@ func infraSpec(projectConfig *ProjectConfig) (*scaffold.InfraSpec, error) {
 	}
 
 	// create reverse frontends -> backends mapping
-	for _, svc := range infraSpec.Services {
+	for i := range infraSpec.Services {
+		svc := &infraSpec.Services[i]
```

Comment:
I like the thinking about atomic guarantees, this is definitely a good thing to think about as another step of increasing "safety", but I don't think this will affect us meaningfully in this particular scenario.